Favourites

MikeW210 opened the pull request:
[add]
user can folow and unfollow other users
[add]
added heart when u follow somebody
[add]
added relationship db
[add]
added axios to display data, to change status if user if following or not
[add]
if user is not logged in he cant follow anybody
[fix]
no weirs frames after clicking follow heart

karczewskiPiotr commented on app/controllers/api/v1/users_controller.rb:
```diff
@@ -2,4 +2,22 @@ class Api::V1::UsersController < ApiController
+    @followers = current_user.following
+  end
+
+  # def update
```
Delete this if not used. :disappointed:

karczewskiPiotr commented on app/javascript/packs/users-table.jsx:
```diff
@@ -1,34 +1,96 @@
+    };
+
+    componentWillMount() {
+        axios.get('/api/v1/users.json', {}, { 'Content-Type': 'application/json' })
```
Use `/api/v1/users` instead of `/api/v1/users.json` because this controller accepts json by default. :wink: 
You could also look into concurrent requests with [axios](https://github.com/axios/axios) but don't feel obliged to.

karczewskiPiotr commented on app/javascript/packs/users-table.jsx:
```diff
@@ -1,34 +1,96 @@
+                this.setState({ users: res.data.data.sort((a,b) => this.compare(a.points,b.points) || this.compare(b.nick.toUpperCase(),a.nick.toUpperCase()) )})
+                
+            });
+            axios.get('/api/v1/users/current/following.json', {}, { 'Content-Type': 'application/json' })
```
Same thing with the `.json` in path.

karczewskiPiotr commented on app/javascript/packs/users-table.jsx:
```diff
@@ -1,34 +1,96 @@
+    }
+
+    click = (user_id) => {
+        axios.post('/api/v1/users/current/following.json', 
```
Here also.

karczewskiPiotr commented on app/javascript/packs/users-table.jsx:
```diff
@@ -1,34 +1,96 @@
+    }
+   
+    render() {
+        let users = this.state.users
```
From what I see these are unnecessary. :open_mouth:

karczewskiPiotr commented on app/views/layouts/application.html.haml:
```diff
@@ -12,6 +12,8 @@
     %p.alert= alert 
     #main
       = render 'layouts/main_nav'
+      
```
Why?

karczewskiPiotr commented on ser.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path fill="none" d="M0 0h24v24H0V0z"/><path d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"/></svg>
```
Is this file used for anything?

karczewskiPiotr commented on spec/models/relationship_spec.rb:
```diff
@@ -0,0 +1,10 @@
+require"rails_helper"
+RSpec.describe Relationship, type:  :model do
+ 
+  describe "relation1" do
```
More descriptive names would be appreciated. :+1:

karczewskiPiotr commented on spec/models/user_spec.rb:
```diff
@@ -22,4 +22,18 @@
+          let(:user1) { create(:user) }
+          let(:user2) { create(:user) }
+        
+        it("should follow and unfollow a user") do
```
Break this into separate `its`

karczewskiPiotr commented on unser.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path fill="none" d="M0 0h24v24H0V0z"/><path d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.8 ...
```
Delete this if not used.

MikeW210 commented:
[add]
added axios to display data, to change status if user if following or not 
[add]
user can follow/unfollow other users
[add]
when not signed in user can't follow/ unfollow 
[add]
added heart as follow sigh added relationship db
[fix]
no weird frames after clicking follow heart
